hwmgr-plugins/metal3/controller/baremetalhost_manager.go

```diff
@@ -595,6 +605,12 @@ func processBMHUpdateCase(ctx context.Context,
 		return fmt.Errorf("unable to initiate update for BMH %s/%s", bmh.Namespace, bmh.Name)
 	}
 
+	// clear transient error annotation if BMH recovered
```

Comment:
This would cover the BMH deletion case as well ? Annotation is set and the node is deleted